### PR TITLE
Add configurable submodule directories; Add tester.py

### PR DESCRIPTION
**Pull request by arturum1**
- Add 'dirs' dictonary in meta_data to allow configuring and overriding directories of submodules instead of having a hard-coded fixed path.
- Update functions to use configurable directories (some of the changes are just reverts of previous commits)
- Merge with IObundle/build-lib
- Add tester.py script
- Update ios.py with new auto-map for interfaces
- Included changes I suggested in my review of https://github.com/IObundle/iob-lib/pull/365

### CODE REVIEW

File: scripts/build_srcs.py [hunk 1]
```diff
@@ -9,13 +9,12 @@
 from submodule_utils import import_setup
 import iob_colors
 
-lib_dir = "./submodules/LIB"
```
Comment (PedroAntunes178) [inline scripts/build_srcs.py:12]: Why did you decide to remove lib_dir directory?
My idea to have the lib_dir directory global was to make it possible to edit this variable if we wanted to overwrite the directory.

File: scripts/build_srcs.py [hunk 2]
```diff
@@ -9,13 +9,12 @@
...
-
 def hw_setup(core_meta_data):
     core_name = core_meta_data['name']
     core_version = core_meta_data['version']
     build_dir = core_meta_data['build_dir']
-    core_dir = core_meta_data['core_dir']
+    core_dir = core_meta_data['setup_dir']
```
Comment (PedroAntunes178) [inline scripts/build_srcs.py]: Why did you choose the name setup_dir?
The setup_dir I interpret it to be the initial directory where the setup.py is called. However, when adding submodules, those should not pass the setup_dir. Instead, they should pass their directory relative to the setup_dir. Which I called core_dir.
What changes did you do to this mechanism?

Comment (PedroAntunes178) [reply scripts/build_srcs.py]: This applies to the changes in https://github.com/PedroAntunes178/iob-soc/pull/3 as well.

File: scripts/build_srcs.py (hunk 1 above)
Comment (arturum1) [reply scripts/build_srcs.py:12]: Re-added global lib_dir in 73999b3

File: scripts/build_srcs.py (hunk 2 above)
Comment (arturum1) [reply scripts/build_srcs.py]: The method I followed for naming these variables was: build_dir is the directory used to build the core, therefore setup_dir is the directory used to setup the core. 

Renamed all 'core_dir' variables to 'setup_dir' in https://github.com/PedroAntunes178/iob-lib/pull/1/commits/73999b3e8d06cb97a703ed281246fef277dc0311

**Pull request merged**